src/axom/primal/tests/primal_nurbs_patch.cpp

```diff
@@ -837,6 +837,172 @@ TEST(primal_nurbspatch, patch_split)
+
+      for(int N = 0; N < DIM; ++N)
+      {
+        EXPECT_NEAR(pt1[N], pt2[N], 1e-10);
+      }
+    }
+  }
+}
+//------------------------------------------------------------------------------
+TEST(primal_nurbspatch, nurbs_parameter_space_scaling)
+{
+  SLIC_INFO("Testing NURBS Patch parameter space expansion");
+
+  const int DIM = 3;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using NURBSPatchType = primal::NURBSPatch<CoordType, DIM>;
+
+  const int npts_u = 5;
+  const int npts_v = 4;
+
+  const int degree_u = 3;
+  const int degree_v = 2;
+
+  // clang-format off
+  PointType controlPoints[5 * 4] = {
+    PointType {0, 0, 0}, PointType {0, 4,  0}, PointType {0, 8, -3}, PointType {0, 12, 0},
+    PointType {2, 0, 3}, PointType {2, 4,  0}, PointType {2, 8,  0}, PointType {2, 12, 0},
+    PointType {4, 0, 0}, PointType {4, 4,  0}, PointType {4, 8,  3}, PointType {4, 12, 0},
+    PointType {6, 0, 0}, PointType {6, 4, -3}, PointType {6, 8,  0}, PointType {6, 12, 0},
+    PointType {8, 0, 0}, PointType {8, 4,  0}, PointType {8, 8,  0}, PointType {8, 12, 0}};
+
+  double weights[5 * 4] = {
+    1.0, 2.0, 3.0, 2.0,
+    2.0, 3.0, 4.0, 3.0,
+    3.0, 4.0, 5.0, 4.0,
+    4.0, 5.0, 6.0, 5.0,
+    5.0, 6.0, 7.0, 6.0};
+  // clang-format on
+
+  NURBSPatchType nPatch(controlPoints, weights, npts_u, npts_v, degree_u, degree_v);
+  NURBSPatchType nPatchTriviallyTrimmed(nPatch);
+  nPatchTriviallyTrimmed.makeTriviallyTrimmed();
+
+  NURBSPatchType supPatch(nPatch);
+  NURBSPatchType supPatchTrimmed(nPatchTriviallyTrimmed);
+
+  // Expand the parameter space of the patch
+  constexpr double scaleFactor = 1.05;
+
+  // Two patches have same untrimmed geometry, but the second is trimmed to the weighty_original boundary
+  supPatch.scaleParameterSpace(scaleFactor);
+  supPatchTrimmed.scaleParameterSpace(scaleFactor);
+
+  double min_u = nPatch.getMinKnot_u();
+  double max_u = nPatch.getMaxKnot_u();
+
+  double min_v = nPatch.getMinKnot_v();
+  double max_v = nPatch.getMaxKnot_v();
+
+  // Check the parameter space of the superpatches' knots
+  EXPECT_NEAR(supPatch.getMinKnot_u(), min_u - (scaleFactor - 1.0), 1e-10);
+  EXPECT_NEAR(supPatch.getMaxKnot_u(), max_u + (scaleFactor - 1.0), 1e-10);
+  EXPECT_NEAR(supPatch.getMinKnot_v(), min_v - (scaleFactor - 1.0), 1e-10);
+  EXPECT_NEAR(supPatch.getMaxKnot_v(), max_v + (scaleFactor - 1.0), 1e-10);
+
+  // Check that the patches are equal in the original parameter space
+  constexpr int npts = 15;
+  double u_pts[npts], v_pts[npts];
+  axom::numerics::linspace(min_u - (scaleFactor - 1.0),
+                           max_u + (scaleFactor - 1.0),
+                           u_pts,
+                           npts);
+  axom::numerics::linspace(min_v - (scaleFactor - 1.0),
+                           max_v + (scaleFactor - 1.0),
+                           v_pts,
+                           npts);
+
+  for(auto u : u_pts)
+  {
+    for(auto v : v_pts)
+    {
```

Comment:
No need to make a change, but sharing in case you haven't seen it. 

We have a templated `ZipIndexable` class that can take arrays of coordinates and treat them as iterable lists of primitives.

e.g. something like the following should work:
```cpp
using PointZipper= primal::ZipIndexable<UVPointType>;
PointZipper it = {{u_pts, v_pts}
for(int i=0; i<npts;++i}
{
  const auto& pt = it[i];
  ...
}
```
(caveat: it might need some minor tweaking, see, e.g. https://github.com/LLNL/axom/blob/develop/src/axom/primal/tests/primal_zip.cpp#L86 )

Also, I was hoping we could use the range-for syntax, but ZipIndexable doesn't know its size :(

Reply:
That's really cool! Probably not worth the headache to get everything changed now, but there will certainly be future tests that could benefit.